File: single-language-analysis/dataset-files/python/test_smell_064.py
import pytest

class TestCellTest:
    def test_smell_064(self):
        assert 0.0 == pytest.approx(Cell.calculate_mean([]), rel=0.001), "Mean of empty list should be 0.0"
        assert 0.0 == pytest.approx(Cell.calculate_mean(None), rel=0.001), "Mean of null should be 0.0"
        assert 2.0 == pytest.approx(Cell.calculate_mean([1.0, 2.0, 3.0]), rel=0.001), "Mean of test numbers should be 2.0"

File: single-language-analysis/dataset-files/java/TestSmell064.java
// Original URL: https://github.com/efleming91/AlternativeLanguage/blob/08766423dcfbf8530e0c5a89eafb2bdfa1db370f/CellTest.java#L75-L80

public class TestSmell064 {
    @Test
    public void testMean() {
        assertEquals("Mean of empty list should be 0.0", 0.0, Cell.calculateMean(new ArrayList<>()), 0.001);
        assertEquals("Mean of null should be 0.0", 0.0, Cell.calculateMean(null), 0.001);
        assertEquals("Mean of test numbers should be 2.0", 2.0, Cell.calculateMean(Arrays.asList(1f, 2f, 3f)), 0.001);
    }
}
